### PR TITLE
Integrate HTTP server

### PR DESCRIPTION
**Pull request by OrMemphis**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -7,14 +7,13 @@ require (
...
 	github.com/go-playground/validator/v10 v10.10.0
-	github.com/gofrs/uuid v4.2.0+incompatible
 	github.com/googollee/go-socket.io v1.6.2
 	github.com/hanzoai/gochimp3 v0.0.0-20210305004051-da66ea724147
 	github.com/klauspost/compress v1.14.4
 	github.com/lightstep/otel-launcher-go v1.5.1
 	github.com/minio/highwayhash v1.0.2
 	github.com/nats-io/jwt/v2 v2.3.0
-	github.com/nats-io/nats.go v1.15.0
+	github.com/nats-io/nats.go v1.16.0
```
Comment (idanasulinmemphis) [inline go.mod:16]: do we still need the nats SDK package?

File: go.sum [hunk 2]
```diff
@@ -292,8 +292,8 @@ github.com/nats-io/jwt/v2 v2.3.0 h1:z2mA1a7tIf5ShggOFlR1oBPgd6hGqcDYsISxZByUzdI=
 github.com/nats-io/jwt/v2 v2.3.0/go.mod h1:0tqz9Hlu6bCBFLWAASKhE5vUA4c24L9KPUUgvwumE/k=
 github.com/nats-io/nats-server/v2 v2.8.4 h1:0jQzze1T9mECg8YZEl8+WYUXb9JKluJfCBriPUtluB4=
 github.com/nats-io/nats-server/v2 v2.8.4/go.mod h1:8zZa+Al3WsESfmgSs98Fi06dRWLH5Bnq90m5bKD/eT4=
-github.com/nats-io/nats.go v1.15.0 h1:3IXNBolWrwIUf2soxh6Rla8gPzYWEZQBUBK6RV21s+o=
-github.com/nats-io/nats.go v1.15.0/go.mod h1:BPko4oXsySz4aSWeFgOHLZs3G4Jq4ZAyE6/zMCxRT6w=
+github.com/nats-io/nats.go v1.16.0 h1:zvLE7fGBQYW6MWaFaRdsgm9qT39PJDQoju+DS8KsO1g=
```
Comment (idanasulinmemphis) [inline go.sum:295]: same here - I think we can delete the use of the nats SDK

File: handlers/handlers.go [hunk 3]
```diff
@@ -187,7 +196,7 @@ func CreateDefaultStation(stationName string, username string) (models.Station,
 		Functions:       []models.Function{},
 	}
 
-	err = broker.CreateStream(newStation)
+	err = s.CreateStation(newStation)
```
Comment (idanasulinmemphis) [inline handlers/handlers.go:199]: change to create stream

File: handlers/monitoring.go [hunk 4]
```diff
@@ -98,19 +93,19 @@ func (mh MonitoringHandler) GetSystemComponents() ([]models.SystemComponent, err
...
-				ActualPods:  1,
-			})
-		} else {
-			components = append(components, models.SystemComponent{
-				Component:   "broker",
-				DesiredPods: 1,
-				ActualPods:  0,
-			})
-		}
+		// if broker.IsConnectionAlive() {
```
Comment (idanasulinmemphis) [inline handlers/monitoring.go:96]: you can remove these lines, in line 126 change the "control-plane" to "broker"

File: handlers/poison_messages.go [hunk 5]
```diff
@@ -15,24 +15,23 @@ package handlers
...
 	"go.mongodb.org/mongo-driver/bson/primitive"
 	"go.mongodb.org/mongo-driver/mongo"
 	"go.mongodb.org/mongo-driver/mongo/options"
 )
 
-type PoisonMessagesHandler struct{}
+type PoisonMessagesHandler struct{ S *server.Server }
 
-func (pmh PoisonMessagesHandler) HandleNewMessage(msg *nats.Msg) {
+func (pmh PoisonMessagesHandler) HandleNewMessage(subject string, msg []byte) {
```
Comment (idanasulinmemphis) [inline handlers/poison_messages.go:32]: first param is not used you can switch to _ instead

File: handlers/stations.go [hunk 6]
```diff
@@ -284,7 +284,7 @@ func (sh StationsHandler) CreateStation(c *gin.Context) {
 		IsDeleted:       false,
 	}
 
-	err = broker.CreateStream(newStation)
+	err = sh.S.CreateStation(newStation)
```
Comment (idanasulinmemphis) [inline handlers/stations.go:287]: change to create stream

File: handlers/stations.go [hunk 7]
```diff
@@ -588,10 +588,23 @@ func (sh StationsHandler) GetMessageDetails(c *gin.Context) {
 		return
 	}
 
-	natsMsg, err := broker.GetMessage(stationName, uint64(body.MessageSeq))
+	sm, err := sh.S.GetMessage(stationName, uint64(body.MessageSeq))
```
Comment (idanasulinmemphis) [inline handlers/stations.go:591]: what sm means?

File: http_server/routes/routes.go [hunk 8]
```diff
@@ -31,12 +32,12 @@ func (lw loggerWriter) Write(p []byte) (int, error) {
...
 	if statusCode != "200" && statusCode != "204" {
-		logger.Error(log)
+		fmt.Errorf(log)
 	}
 	return len(p), nil
 }
 
-func InitializeHttpRoutes() *gin.Engine {
+func InitializeHttpRoutes(handlers *handlers.Handlers) *gin.Engine {
 	gin.DefaultWriter = loggerWriter{}
```
Comment (idanasulinmemphis) [inline http_server/routes/routes.go:41]: we can remove this line + the loggerWriter + write ffunction, we don't need this anymore

File: main.go [hunk 9]
```diff
@@ -100,6 +101,10 @@ func handleError(s *server.Server, message string, err error) {
 
 func runMemphis(s *server.Server) {
 
+	if !s.MemphisInitialized() {
+		s.Fatalf("Jetstream not enabled on global account")
```
Comment (idanasulinmemphis) [inline main.go:105]: what this error means

File: server/memphis_helper.go [hunk 10]
```diff
@@ -0,0 +1,478 @@
...
+
+func AddUser(username string) (string, error) {
+	return configuration.CONNECTION_TOKEN, nil
+}
+
+func RemoveUser(username string) error {
+	return nil
+}
+
+func CreateProducer() error {
```
Comment (idanasulinmemphis) [inline server/memphis_helper.go:42]: we can remove this function and the use of it across the entire application

File: server/memphis_helper.go [hunk 11]
```diff
@@ -0,0 +1,478 @@
...
+	// nothing to create
+	return nil
+}
+
+func RemoveProducer() error {
```
Comment (idanasulinmemphis) [inline server/memphis_helper.go:47]: we can remove this function and the use of it across the entire application

File: server/memphis_helper.go [hunk 12]
```diff
@@ -0,0 +1,478 @@
...
+	// nothing to remove
+	return nil
+}
+
+func (s *Server) CreateStation(station models.Station) error {
```
Comment (idanasulinmemphis) [inline server/memphis_helper.go:52]: change to create stream

File: server/memphis_helper_test.go [hunk 13]
```diff
@@ -0,0 +1,112 @@
+// Copyright 2019-2022 The NATS Authors
```
Comment (idanasulinmemphis) [inline server/memphis_helper_test.go:1]: please replace those headers to Memphis headers across the entire application. You can find the Memphis headers inside 1 of the memphis files from the staging branch

File: go.mod (hunk 1 above)
Comment (OrMemphis) [reply go.mod:16]: yes, NATS' tests use it, and they also rely on latest version

File: handlers/stations.go (hunk 7 above)
Comment (OrMemphis) [reply handlers/stations.go:591]: StoredMsg, following the terms they use inside NATS.
should I change anyway?

File: main.go (hunk 9 above)
Comment (OrMemphis) [reply main.go:105]: that we started NATS w/o enabling Jetstream, 
mostly internal as we should always use NATS+JS, 
a good point, we should make the default behaviour to run with the --js flag

**Review by idanasulinmemphis — APPROVED** [review]

**Pull request merged**